namui-cli/src/services/bundle_metadata_service.rs

```diff
@@ -0,0 +1,130 @@
+use crate::util::get_namui_bundle_manifest;
+use std::{collections::HashMap, fs, path::PathBuf, sync::RwLock};
+
+pub struct BundleMetadataService {
+    bundle_metadata_string: RwLock<String>,
+    url_src_path_map: RwLock<HashMap<PathBuf, PathBuf>>,
+}
+
+impl BundleMetadataService {
+    pub fn new() -> Self {
+        BundleMetadataService {
+            bundle_metadata_string: RwLock::new(String::new()),
+            url_src_path_map: RwLock::new(HashMap::new()),
+        }
+    }
+
+    pub fn load_bundle_manifest(&self, project_root_path: &PathBuf) -> Result<(), String> {
+        let namui_bundle_manifest = get_namui_bundle_manifest(project_root_path)?;
+        let src_dest_path_map = namui_bundle_manifest.query(project_root_path, &PathBuf::new())?;
+        let url_src_path_map: HashMap<PathBuf, PathBuf> =
+            convert_src_dest_path_map_to_url_src_path_map(src_dest_path_map);
+
+        let url_list: Vec<&PathBuf> = url_src_path_map.keys().collect();
+        let bundle_metadata_string = serde_json::to_string(&url_list).map_err(|error| {
```

Comment:
is it right to use map_err here? not return Err?

Reply:
Convert `serde_json::Error` to `String` with map_err, then return Err with question mark operator